### PR TITLE
Merge the 1.19 Work.

### PR DESCRIPTION
**Pull request by wolflord212312**
This is a 1.19 Version Of the Forge-client-reset ,that [adde010](https://github.com/adde0109) Made,Could we get this merged and put on Curseforge???

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,26 +2,17 @@ subprojects {
     apply plugin: 'java'
     apply plugin: 'maven-publish'
 
-    group = 'gg.chaldea.client.reset.packet'
-    version = '0.0.5'
+    group = 'net.foxmcloud.clientresetpacket'
```
Comment (danorris709) [inline build.gradle:5]: This shouldn't be changed

File: build.gradle [hunk 2]
```diff
@@ -2,26 +2,17 @@ subprojects {
...
-    targetCompatibility = 1.8
+    java.toolchain.languageVersion = JavaLanguageVersion.of(17)
 
     tasks.withType(JavaCompile) {
         options.encoding = 'UTF-8'
     }
 
     repositories {
         mavenCentral()
-        maven {
```
Comment (danorris709) [inline build.gradle:17]: Why did you remove this

File: forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java [hunk 3]
```diff
@@ -1,116 +1,185 @@
...
+				handshakeChannel.messageBuilder(S2CReset.class, 98)
+						.loginIndex(S2CReset::getLoginIndex, S2CReset::setLoginIndex)
+						.decoder(S2CReset::decode)
+						.encoder(S2CReset::encode)
+						.consumerNetworkThread(HandshakeHandler.biConsumerFor(ClientReset::handleReset))
+						.add();
+				logger.info(RESETMARKER, "Registered forge reset packet successfully.");
+			}
+		}
+		catch (Exception e) {
```
Comment (danorris709) [inline forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java:84]: What is this actually catching? The catch should be more specific and refactored out so it doesn't cover code it doesn't need to

File: forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java [hunk 4]
```diff
@@ -1,116 +1,185 @@
...
+		}
+		if (contextConstructor == null) {
+			logger.error(RESETMARKER, "Failed to find FML's network event context constructor. Disabling mod.");
+			return;
+		}
+		try {
+			//handshakeField.setAccessible(true);
+			//contextConstructor.setAccessible(true);
+			Object handshake = handshakeField.get(null);
+			if (handshake instanceof SimpleChannel) {
```
Comment (danorris709) [inline forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java:72]: Why would we not throw an error if it's not an instance of simple channel, because it won't throw an exception so the catch clause would not be reached

File: forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java [hunk 5]
```diff
@@ -1,116 +1,185 @@
...
+			return;
+		}
+		if (contextConstructor == null) {
+			logger.error(RESETMARKER, "Failed to find FML's network event context constructor. Disabling mod.");
+			return;
+		}
+		try {
+			//handshakeField.setAccessible(true);
+			//contextConstructor.setAccessible(true);
+			Object handshake = handshakeField.get(null);
```
Comment (danorris709) [inline forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java:71]: Why is it now using reflection and not access transformers like in 1.16.5? It should be consistent, and using ATs

File: forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java [hunk 6]
```diff
@@ -1,116 +1,185 @@
...
+		));
+
+		context.setPacketHandled(true);
+		try {
+			handshakeChannel.reply(
+				new HandshakeMessages.C2SAcknowledge(),
+				(NetworkEvent.Context)contextConstructor.newInstance(connection, NetworkDirection.LOGIN_TO_CLIENT, 98)
+			);
+		}
+		catch (Exception e) {
```
Comment (danorris709) [inline forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java:117]: Again, what is this actually catching?

File: forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java [hunk 7]
```diff
@@ -1,116 +1,185 @@
...
+
+			// Clear
+			if (Minecraft.getInstance().level == null) {
+				// Ensure the GameData is reverted in case the client is reset during the handshake.
+				GameData.revertToFrozen();
+			}
+			Minecraft.getInstance().getClientPackSource().serverPack = null;
+
+			// Clear
+			Minecraft.getInstance().clearLevel(new GenericDirtMessageScreen(Component.literal("Negotiating..."/*"connect.negotiating"*/)));
```
Comment (danorris709) [inline forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java:142]: Don't remove lang keys

File: forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java [hunk 8]
```diff
@@ -1,116 +1,185 @@
...
+
+			// Restore
+			Minecraft.getInstance().getClientPackSource().serverPack = serverPack;
```
Comment (danorris709) [inline forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java:145]: Is this actually doing anything? I believe that the variable only gets set in one method which then calls to reload the resource pack so setting this variable without reloading the resource pack either won't do anything, or potentially cause issues?

**Review by danorris709 — CHANGES_REQUESTED** [review]
Firstly, if you want to add a second version to the repository we should rename the "forge" directory to "forge-1.16.5" and then do the second version in a second directory called "forge-{version}" so we can keep both versions and update as needed.

Secondly, you need to make sure all your code follows [Google's Style Guide](https://google.github.io/styleguide/javaguide.html)

Third, you've completely changed how this functions from using access transformers to using reflection which you shouldn't have done.

Finally, there was an open PR for this on Forge's latest branch (1.19.4) [here](https://github.com/MinecraftForge/MinecraftForge/pull/9313) although closed (seemingly for no reason? wasn't even discussed in their Discord as far as I can see) the comments on there from the one reviewer are actually quite useful and I think we should follow them more closely. They are correct about removing the lines that are removed (here via mixin) and the potential issues it causes. Perhaps this requires more packets in this mod and is outside of the scope of this PR but still noteworthy

**Comment by wolflord212312** [comment]
I will try to get these fixed,But it will take a while lol But for now i will close this ,thank you for your input
